crates/shielded_token/src/vp.rs

```diff
@@ -624,15 +624,33 @@ where
                 return Err(error);
             }
         }
+        let masp_tx_id = shielded_tx.txid().into();
+        let has_trans_inputs = shielded_tx
+            .transparent_bundle()
+            .map(|b| !b.vin.is_empty())
+            .unwrap_or(false);
         // The transaction shall not push masp authorizer actions that are not
-        // needed cause this might lead vps to run a wrong validation logic
-        if !actions_authorizers.is_empty() {
+        // needed because this might lead vps to run a wrong validation logic
+        if !actions_authorizers.is_empty() && !has_trans_inputs {
```

Comment:
Here we might have an issue: if the tx has transparent inputs then we might miss the fact that some of the required authorizers were not actually needed

Reply:
I think you're right. I will try to fix the above block. I